### PR TITLE
Add the progress bar for the DataStore

### PR DESCRIPTION
**Pull request by bkhelifi**
**Description**
This pull request add the progress bar in the function `get_observations`.

**Dear reviewer**
I am wondering if, in general, one should mention in the docstring that a progress bar is possible for a function...

### CODE REVIEW

File: gammapy/data/obs_table.py [hunk 1]
```diff
@@ -74,7 +74,12 @@ def select_obs_id(self, obs_id):
 
     def summary(self):
         """Summary info string (str)."""
-        obs_name = self.meta.get("OBSERVATORY_NAME", "N/A")
+        obs_name = self.meta.get(
+            "OBSERVATORY_NAME", "N/A"
+        )  # This is not GADF compliant
+        if "N/A" in obs_name:
+            obs_name = self.meta.get("OBSERVER", "N/A")
```
Comment (AtreyeeS) [inline gammapy/data/obs_table.py:81]:
```suggestion
            obs_name = self.meta.get("OBSERVATORY_NAME", self.meta.get("OBSERVER", "N/A"))
```

**Review by AtreyeeS — APPROVED** [review]
Thanks @bkhelifi !

File: gammapy/data/obs_table.py [hunk 2]
```diff
@@ -74,7 +74,12 @@ def select_obs_id(self, obs_id):
 
     def summary(self):
         """Summary info string (str)."""
-        obs_name = self.meta.get("OBSERVATORY_NAME", "N/A")
+        obs_name = self.meta.get(
```
Comment (AtreyeeS) [inline gammapy/data/obs_table.py:77]: Lines 77-80 are not necessary :-)

**Review by Astro-Kirsty — APPROVED** [review]
Look good from my side @bkhelifi

File: gammapy/data/obs_table.py [hunk 3]
```diff
@@ -74,7 +74,12 @@ def select_obs_id(self, obs_id):
 
     def summary(self):
         """Summary info string (str)."""
-        obs_name = self.meta.get("OBSERVATORY_NAME", "N/A")
```
Comment (registerrier) [inline gammapy/data/obs_table.py:77]: I tend to agree with @AtreyeeS . If there is no keyword specification for the observatory name in the specifications, we shouldn't try to guess one. 
I would competely remove this part.

**Review by registerrier — COMMENTED** [review]
Thanks @bkhelifi . See small suggestion.

**Review by QRemy — COMMENTED** [review]
I don't think this is useful because `get_observations` does not read the files as we use lazy loading so it should be very fast and the display time of the progress bar probably add too much overhead.

**Comment by bkhelifi** [comment]
The lazy loading is set to False since its introduction...
On my computer, this function takes more than 4 min, which I realised by introducing this optional feature.
It costs nothing and it is useful;)

File: gammapy/data/obs_table.py (hunk 3 above)
Comment (bkhelifi) [reply gammapy/data/obs_table.py:77]: Agree, even if the gammapy team and friends are leading the data formatting. It is not 'others' that will make specifications.

**Comment by registerrier** [comment]
> The lazy loading is set to False since its introduction... On my computer, this function takes more than 4 min, which I realised by introducing this optional feature. It costs nothing and it is useful;)

It is the caching of the loaded content that is set to `False` in the lazy loading mechanism. So we re-read from disk each time we access one of the HDU. But when you extract the observation from the datastore, its content is not yet loaded.  

Technically the operation `get_observations()` is long only when you extract a very large number of observations from the `DataStore` (typically >10000).

**Comment by QRemy** [comment]
> So we re-read from disk each time we access one of the HDU

Does that mean we also re-decompress the whole file each time we access one of the HDU ? (if yes that probably takes most of the time)

**Comment by morcuended** [comment]
> Possibly I am mistaken, but did you not face similarly large times to load observations for the LST-1 dataset @morcuended ?

Yes, I think the problem was the unzipping of the files. We reduced the run time from the order of 30 mins to just several minutes when using already unzipped files.

**Review by QRemy — APPROVED** [review]
Thanks, no further comments.
I corrected the formatting and will merge now.

**Pull request merged**